server/src/main/java/io/crate/expression/reference/StaticTableDefinition.java

```diff
@@ -59,8 +60,15 @@ public StaticTableDefinition(Supplier<CompletableFuture<? extends Iterable<T>>>
     public StaticTableDefinition(Supplier<? extends Iterable<T>> iterable,
                                  BiPredicate<Role, T> predicate,
                                  Map<ColumnIdent, ? extends RowCollectExpressionFactory<T>> expressionFactories) {
+        this(iterable, predicate, (u, t) -> t, expressionFactories);
+    }
+
+    public StaticTableDefinition(Supplier<? extends Iterable<T>> iterable,
+                                 BiPredicate<Role, T> predicate,
+                                 BiFunction<Role, T, T> mapper,
+                                 Map<ColumnIdent, ? extends RowCollectExpressionFactory<T>> expressionFactories) {
         this.getRecords = (txnCtx, u) -> completedFuture(() -> StreamSupport.stream(iterable.get().spliterator(), false)
-            .filter(t -> u == null || predicate.test(u, t)).iterator());
+            .filter(t -> u == null || predicate.test(u, t)).map(t -> mapper.apply(u, t)).iterator());
```

Comment:
Couldn't this be done on the call-site of the constructor instead of adding a new overload? `StaticTableDefinition` already has so many...